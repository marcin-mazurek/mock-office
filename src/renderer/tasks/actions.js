export const ADD = 'tasks/ADD';
export const REMOVE = 'tasks/REMOVE';

export const add = (task, taskId) => ({
  type: ADD,
  task,
  taskId
});

export const remove = id => ({
  type: REMOVE,
  id
});
