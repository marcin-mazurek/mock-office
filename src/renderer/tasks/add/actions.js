export const INIT = 'addTask/INIT';

export const init = (serverId, task) => ({
  type: INIT,
  serverId,
  task
});
