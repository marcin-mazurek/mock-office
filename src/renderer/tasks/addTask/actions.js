export const INIT = 'addTask/INIT';
export const ADD = 'tasks/ADD';

export const init = (serverId, tasks) => ({
  type: INIT,
  serverId,
  tasks
});

export const add = (serverId, taskId, taskPayload, path = 'all paths') => ({
  type: ADD,
  serverId,
  taskId,
  taskPayload,
  path
});
