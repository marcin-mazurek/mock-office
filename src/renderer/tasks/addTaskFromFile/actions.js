export const INIT = 'pickTaskFile/INIT';

export const init = (serverId, files) => ({
  type: INIT,
  serverId,
  files
});
