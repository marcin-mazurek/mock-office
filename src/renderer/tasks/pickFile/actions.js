export const INIT = 'pickTaskFile/INIT';

export const init = files => ({
  type: INIT,
  files
});
