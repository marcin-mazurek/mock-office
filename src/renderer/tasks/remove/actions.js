export const INIT = 'remove/init';

export const init = (queueId, taskId) => ({
  type: INIT,
  taskId,
  queueId
});
