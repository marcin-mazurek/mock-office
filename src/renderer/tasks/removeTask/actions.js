export const INIT = 'remove/init';
export const REMOVE = 'tasks/REMOVE';

export const init = (queueId, taskId) => ({
  type: INIT,
  taskId,
  queueId
});

export const remove = (queueId, taskId) => ({
  type: REMOVE,
  queueId,
  taskId
});
