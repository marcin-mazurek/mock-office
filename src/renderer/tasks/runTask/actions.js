export const RUN = 'tasks/RUN';

export const run = (serverId, taskId) => ({
  type: RUN,
  serverId,
  taskId
});
