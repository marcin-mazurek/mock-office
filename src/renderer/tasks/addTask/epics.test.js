import { Map, Set } from 'immutable';
import configureMockStore from 'redux-mock-store';
import 'rxjs';
import { createEpicMiddleware } from 'redux-observable';
import { init } from './actions';

const mockedElectron = {
  remote: {
    require() {
      return {
        default: {
          find() {
            return {
              addTask() {
                return 'task id';
              }
            };
          }
        }
      };
    }
  }
};

jest.mock('electron', () => mockedElectron);
const addTaskEpic = require('./epics').default;

let store;
const epicMiddleware = createEpicMiddleware(addTaskEpic);
const mockStore = configureMockStore([epicMiddleware]);

test('addTaskEpic', () => {
  store = mockStore(new Map({
    servers: new Map({
      entities: new Map({
        'server id': {
          type: 'server type',
          name: 'server name',
          port: 3000,
          id: 'server id'
        }
      }),
      selected: 'server id',
      running: new Set()
    }),
    tasks: new Map()
  }));

  store.dispatch(init('serverId', [
    {
      taskPayload: 'taskPayload'
    }
  ]));
  expect(store.getActions()).toMatchSnapshot();
});
