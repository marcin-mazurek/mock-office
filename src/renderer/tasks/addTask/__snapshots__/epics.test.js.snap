exports[`test addTaskEpic 1`] = `
Array [
  Object {
    "serverId": "serverId",
    "tasks": Array [
      Object {
        "requirements": Object {
          "url": "/some-path",
        },
        "taskPayload": "taskPayload",
      },
    ],
    "type": "addTask/INIT",
  },
  Object {
    "path": "/some-path",
    "serverId": "serverId",
    "taskId": "task id",
    "taskPayload": "taskPayload",
    "type": "tasks/ADD",
  },
]
`;
